rest-api/src/main/java/gov/cms/qpp/conversion/api/controllers/v1/QrdaControllerV1.java

```diff
@@ -17,13 +20,24 @@
 @RequestMapping(path = "/", headers = {"Accept=" + Constants.V1_API_ACCEPT})
 public class QrdaControllerV1 extends SkeletalQrdaController<String> {
 
+	/**
+	 * Constructor to super class to initialize fields
+	 * @param qrdaService {@link QrdaService} to perform QRDA to QPP conversion
+	 * @param validationService {@link ValidationService} to perform post conversion validation
+	 * @param auditService {@link AuditService} to persist audit information
+	 */
 	public QrdaControllerV1(QrdaService qrdaService, ValidationService validationService, AuditService auditService) {
 		super(qrdaService, validationService, auditService);
 	}
 
 	@Override
-	protected String respond(ConversionReport report) {
-		return report.getEncoded().toString(); // TODO toObject?
+	protected String respond(MultipartFile file, String checkedPurpose, HttpHeaders httpHeaders) {
+		ConversionReport conversionReport = buildReport(file.getOriginalFilename(), inputStream(file), checkedPurpose);
+		Metadata metadata = audit(conversionReport);
+		if (null != metadata) {
+			httpHeaders.add("Location", metadata.getUuid());
+		}
```

Comment:
Should we check for null != metadata.getUuid() ?
(I like the null first as you have it here though I only recently adopted the pattern myself and often forget to do it.)